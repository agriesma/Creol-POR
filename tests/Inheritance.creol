
class A(a: Integer)
begin
  op init == skip
  op run == skip
end

class B(b: Integer) inherits A(b)
begin
  op init == prove a = b
  op run == skip
end
