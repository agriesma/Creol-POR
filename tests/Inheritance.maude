load creol-interpreter
mod PROGRAM is
pr CREOL-INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< "A" : Cl | Inh: noInh, Par: "a", Att: "a" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) >, Ocnt: 0 >

< "B" : Cl | Inh: "A" < "b" > , Par: "b", Att: "b" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) >, Ocnt: 0 >

.
endm
