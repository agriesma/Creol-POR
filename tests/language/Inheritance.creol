/*
 * Inheritance.creol
 *
 * Test the initialisation order.
 *
 * Execute the test with creating one instance of B(1), i.e.:
 *
 * main("B", int(1))
 */
class A(a: Int)
begin
  op init == skip
  op run == skip
end

class B(b: Int) inherits A(b)
begin
  op init == assert a = b
  op run == skip
end
