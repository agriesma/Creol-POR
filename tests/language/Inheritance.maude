load creol-interpreter
mod PROGRAM is
protecting CREOL-SIMULATOR .
op classes : -> Configuration [ctor] .
eq classes =
< "A" : Cl | Inh: noInh, Par: "a", Att: "a" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) >, Ocnt: 0 >

< "B" : Cl | Inh: "A" < "b" >, Par: "b", Att: "b" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) >, Ocnt: 0 >

.
endm
