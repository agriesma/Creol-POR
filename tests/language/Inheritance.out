==========================================
reduce in PROGRAM : init .
result Configuration: 
< "A" : Cl | Inh: noInh, Par: "a", Att: "a" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >,
  Ocnt: 0 >

< "B" : Cl | Inh: "A" < "b" >, Par: "b", Att: "b" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >,
  Ocnt: 0 >
==========================================
rewrite in PROGRAM : init
main("B", int(1)) .
result Configuration: (comp(label(ob("B0"),2), emp) from ob("B0") to ob("B0")
)

< "A" : Cl | Inh: noInh, Par: "a", Att: "a" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >,
  Ocnt: 0 >

< "B" : Cl | Inh: "A" < "b" >, Par: "b", Att: "b" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >,
  Ocnt: 1 >

< ob("B0") : "B" |
  Att: "a" |-> int(1), "b" |-> int(1), "this" |-> ob("B0"),
  Pr: idle,
  PrQ: noProc,
  Dealloc: label(ob("B0"),2),
  Ev: noMsg,
  Lcnt: 3 >

< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >
Bye.
